lib/mix/lib/mix/dep/loader.ex

```diff
@@ -320,6 +322,13 @@ defmodule Mix.Dep.Loader do
     end
   end
 
+  defp incompatible_requirement?(%Mix.Dep{opts: opts}) do
+    case opts[:lock] do
+      {scm, _, _, _} -> is_nil(opts[scm])
```

Comment:
Thanks for the feedback.

After looking around, I think that the patterns for the lock are

1. `{:hex, _, _}` for hex
2. `{scm, _, _, _}` for git
3. `"VERSION"` I am not sure what this one is, but it came out when I ran the tests.

This function is meant to return false when the current dependency and the lock are not of the same kind, e.g. the dependency was using git but now uses hex, or vice-versa. It only works for the first case now. Would making this work for the other cases be good enough, or is this there a better approach for this?